internal/locate/region_request.go

```diff
@@ -1560,6 +1649,14 @@ func (s *RegionRequestSender) onSendFail(bo *retry.Backoffer, ctx *RPCContext, e
 		return errors.WithStack(err)
 	} else if LoadShuttingDown() > 0 {
 		return errors.WithStack(tikverr.ErrTiDBShuttingDown)
+	} else if errors.Cause(err) == context.DeadlineExceeded && req.MaxExecutionDurationMs < 1000 {
+		if s.replicaSelector != nil {
+			// TODO(crazycs520): need discuss.
+			if accessLeader, ok := s.replicaSelector.state.(*accessKnownLeader); ok {
+				s.replicaSelector.state = &tryFollower{leaderIdx: accessLeader.leaderIdx, lastIdx: accessLeader.leaderIdx}
+			}
```

Comment:
I think it's ok to change state for `accessKnownLeader` here, `accessFollower` can make state change in it's next when leader is exhausted.